scripts/azureml-assets/azureml/assets/get_tokens.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Generate SAS tokens for models."""
+
+import argparse
+import re
+from pathlib import Path
+from typing import List
+
+import azureml.assets as assets
+import azureml.assets.util as util
+from azureml.assets.config import AssetType, AzureBlobstoreAssetPath
+
+from collections import defaultdict
+import datetime
+import json
+
+
+def get_tokens(input_dirs: List[Path],
+               asset_config_filename: str,
+               json_output_path: str,
+               pattern: re.Pattern = None):
+    """Generate SAS tokens for models to JSON output file.
+
+    Args:
+        input_dirs (List[Path]): List of directories to search for assets.
+        asset_config_filename (str): Asset config filename to search for.
+        json_output_path (str): Path of JSON file to write output to.
+        pattern (re.Pattern, optional): Regex pattern for assets to copy. Defaults to None.
+    """
+    json_info = defaultdict(dict)
+
+    # Filter to only models
+    for asset_config in util.find_assets(
+            input_dirs, asset_config_filename, types=[AssetType.MODEL], pattern=pattern):
+
+        model_config: assets.ModelConfig = asset_config.extra_config_as_object()
+        path = model_config.path
+
+        if not isinstance(path, AzureBlobstoreAssetPath):
+            continue
+
+        account_name = model_config.path.storage_name
+        container_name = model_config.path.container_name
+
+        if container_name in json_info[account_name]:
+            continue
+
+        _ = model_config.path.get_uri(token_expiration_timedelta=datetime.timedelta(days=1))
+        token = model_config.path.token
```

Comment:
If token is the empty string, convert it to None before it goes into json_info so that in the tokens JSON file None means it's a public container. We don't need a mix of "" and None there when "" is really just used internally to the class to say "I've verified it's public" or maybe comment above) "I broke trying to generate a SAS token".